android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/DecodingAsyncTask.java

```diff
@@ -19,61 +19,75 @@
 import android.os.AsyncTask;
 
 import com.github.barteksc.pdfviewer.source.DocumentSource;
-import com.shockwave.pdfium.PdfDocument;
-import com.shockwave.pdfium.PdfiumCore;
+
+import org.benjinus.pdfium.PdfDocument;
+import org.benjinus.pdfium.PdfiumSDK;
+import org.benjinus.pdfium.util.Size;
+
+import java.lang.ref.WeakReference;
 
 class DecodingAsyncTask extends AsyncTask<Void, Void, Throwable> {
 
     private boolean cancelled;
 
-    private PDFView pdfView;
 
     private Context context;
-    private PdfiumCore pdfiumCore;
-    private PdfDocument pdfDocument;
+    private PdfiumSDK pdfiumSDK;
+
     private String password;
     private DocumentSource docSource;
-    private int firstPageIdx;
-    private int pageWidth;
-    private int pageHeight;
 
-    DecodingAsyncTask(DocumentSource docSource, String password, PDFView pdfView, PdfiumCore pdfiumCore, int firstPageIdx) {
+    private int[] userPages;
+    private WeakReference<PDFView> pdfViewReference;
+    private PdfFile pdfFile;
+
+    DecodingAsyncTask(DocumentSource docSource, String password, int[] userPages, PDFView pdfView, PdfiumSDK pdfiumSDK) {
         this.docSource = docSource;
-        this.firstPageIdx = firstPageIdx;
+        this.userPages = userPages;
         this.cancelled = false;
-        this.pdfView = pdfView;
+        this.pdfViewReference = new WeakReference<>(pdfView);

```

Comment:
They were changing it when changing the engine. Maybe they tried to avoid this leak? 

<img width="514" alt="Screenshot 2024-01-18 at 18 14 03" src="https://github.com/VERO-Digital-Solutions/AndroidPdfViewerV2/assets/57250076/1393b625-a6dc-49bc-8344-2ff225623825">